src/DuneTaskProvider.ml

```diff
@@ -90,19 +83,24 @@ let computeTasks cancellationToken =
   in
   Some tasks
 
-let provideTasks =
+let provideTasks toolchain =
  fun [@bs] cancellationToken ->
-  match Settings.get Setting.t with
+  match Settings.get ~section:"ocaml" Setting.t with
   | None
   | Some false ->
     Js.Promise.resolve None
-  | Some true -> computeTasks cancellationToken
+  | Some true -> computeTasks cancellationToken toolchain
 
 let resolveTask =
  fun [@bs] task _cancellationToken -> Js.Promise.resolve (Some task)
 
-let create () =
-  let provider = { TaskProvider.provideTasks; resolveTask } in
-  Tasks.registerTaskProvider ~typ:task_type ~provider
+let create () = ref None
+
+let register t toolchain =
+  let provideTasks = provideTasks toolchain in
+  let provider = TaskProvider.{ provideTasks; resolveTask } in
```

Comment:
Slightly better is `{ TaskProvider.provideTasks ; relativeTasks }`. When it's possible to avoid an open, we should do it.